src/App.js

```diff
@@ -21,45 +21,118 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
   return squares;
 };
 
 const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
+  // ALL PIECES OF STATE
+  // PIECES OF STATE FOR UPDATING SQUARES
   const [squares, setSquares] = useState(generateSquares());
+  // PIECE OF STATE FOR PLAYER TURNS
+  const [turn] = useState({ playerTurn: false});
+  // PIECE OF STATE FOR DETERMINING THE WINNER
+  const [winner, setWinner] = useState(null);
+  
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  
+  const onClickCallback = (id) => {
+    const updateSquare = squares.map((square) => {
+      for (let subSquare of square) {
+        if (subSquare.id === id && subSquare.value === '' && !winner) {
+          subSquare.value = getCurrentPlayer();
+          if (checkForWinner() !== null) {
+            setWinner(checkForWinner());
+          }
+        }
+      }
+      return square;
+    });
+    setSquares(updateSquare); 
+  };
 
-  const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if
-    //    all three squares have the same value.
+  const getCurrentPlayer = () => {
+    turn.playerTurn = !turn.playerTurn;
+    
+    if (turn.playerTurn) {
+      console.log(playerOne);
+      return playerOne;
+    } else {
+      console.log(playerTwo);
+      return playerTwo;
+    }
   };
```

Comment:
with the suggested state I added at the top you could do a check like 
```suggestion
  const getCurrentPlayer = () => {
          currentSquare.value = currentPlayer;
          if (currentPlayer === PLAYER_1) {
            setCurrentPlayer(PLAYER_2);
          } else {
            setCurrentPlayer(PLAYER_1);
          }
  };
```